File: src/main/java/pl/koziolekweb/money/basics/RoundingTraps.java
package pl.koziolekweb.money.basics;

/**
 * User: koziolek
 */
public class RoundingTraps {

	public static void main(String[] args) {
		double[] numbers = new double[] { 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9 };
		for (double nb : numbers) {
			System.out.format("1 + nb - nb == nb - nb + 1 dla %s  to %s\n",
					nb, 1 + nb - nb == nb - nb + 1);
		}
	}
}

File: src/main/java/pl/koziolekweb/money/basics/rounding_traps.erl
%%%-------------------------------------------------------------------
%%% @author koziolek
%%% @copyright (C) 2015, <COMPANY>
%%% @doc
%%%
%%% @end
%%% Created : 18. mar 2015 20:02
%%%-------------------------------------------------------------------
-module(rounding_traps).
-author("koziolek").
%% API
-export([trap/0]).

trap() ->
  Numbers = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9],
  lists:foreach(fun(Nb) ->
    io:format("~w ~w ~n", [Nb, 1 + Nb - Nb == Nb - Nb + 1])
  end,
    Numbers).
